pkg/render/node.go

```diff
@@ -492,6 +536,51 @@ func (c *nodeComponent) nodeRole() *rbacv1.ClusterRole {
 	return role
 }
 
+// cniPluginRole creates the role containing policy rules that allow the Calico CNI plugin to operate normally.
+func (c *nodeComponent) cniPluginRole() *rbacv1.Role {
+	finalizer := []string{}
+	if !c.cfg.Terminating {
+		finalizer = []string{NodeFinalizer}
+	}
+	role := &rbacv1.Role{
+		TypeMeta: metav1.TypeMeta{Kind: "Role", APIVersion: "rbac.authorization.k8s.io/v1"},
```

Comment:
I think this needs to be a cluster role? The CNI plugin operates on resources across multiple namespaces.

The binding can still be a RoleBinding, though.

Reply:
In your other PR, this is a ClusterRole: https://github.com/projectcalico/calico/pull/7106/files#diff-a13e3dc70d98c53fce0f3a76fa93787186a8d68d182bc2ae231f45afd5d6b365R182